pkg/issuer/acme/dns/util/wait.go

```diff
@@ -77,7 +78,12 @@ func updateDomainWithCName(r *dns.Msg, fqdn string) string {
 }
 
 // checkDNSPropagation checks if the expected TXT record has been propagated to all authoritative nameservers.
-func checkDNSPropagation(fqdn, value string, nameservers []string) (bool, error) {
+func checkDNSPropagation(fqdn, value string, nameservers []string,
+	useAuthoritative bool) (bool, error) {
+	if !useAuthoritative {
+		return checkAuthoritativeNss(fqdn, value, nameservers)
```

Comment:
I think this block needs to actually exist after the CNAME update block just after it, else CNAME'd DNS names won't be resolved if `useAuthoritative == false` (unless by performing the recursive TXT lookup it'll automatically follow the CNAME?)?

Reply:
Nice pickup, was trying to think of way to make a test but I am at a loss. Any thoughts or move on?